Script to update vSphere credential (#9189)
aws/eks-anywhere: docs/content/en/docs/clustermgmt/cluster-upgrades/vsphere-credential-update.md

    - `vsphere-credentials` under `eksa-system` namespace - Update `password`, `passwordCP`, `passwordCSI` field under data.
    - If annotation `kubectl.kubernetes.io/last-applied-configuration` exists on any of the above Secret object, update password in `kubectl.kubernetes.io/last-applied-configuration` field.
    -  `{CLUSTER_NAME}-csi-vsphere-config` under `eksa-system` namespace - If annotation `kubectl.kubernetes.io/last-applied-configuration` exists on the secret object, update password in `kubectl.kubernetes.io/last-applied-configuration` field.

> - Alternatively you can use the vSphere credential update script below to update vSphere credential. Before running the script, please make sure that

Reviewer: can we have this as a separate option? Like option1 - Manual steps, option2 - script that you created, option3 - run upgrade?

> - Alternatively you can use the vSphere credential update script below to update vSphere credential. Before running the script, please make sure that

Reviewer: can we have this as a separate option? Like option1 - Manual steps, option2 - script that you created, option3 - run upgrade?

    - `KUBECONFIG` environment variable has been set to the kubeconfig file generated by EKS-A.

>     - Both `EKSA_VSPHERE_USERNAME` and `EKSA_VSPHERE_PASSWORD` environment variables are set and `EKSA_VSPHERE_PASSWORD` has been updated to the new credential.

Reviewer: ```suggestion
    - Set `EKSA_VSPHERE_USERNAME` and `EKSA_VSPHERE_PASSWORD` with new credentials as environment variables.
```

> The script will need the name of the EKS Anywhere cluster and the name of vSphere server, `./update_vsphere_creds.sh CLUSTER_NAME VSPHERE_SERVER_NAME`

Reviewer: ```suggestion
Save below script and run it by passing EKS Anywhere cluster name and the name of vSphere server, `./update_vsphere_creds.sh CLUSTER_NAME VSPHERE_SERVER_NAME`
```

2. Only update Secret `vsphere-credentials` under `eksa-system` namespace then trigger a full EKS-A CAPI cluster upgrade by modifying the cluster spec:
- Update `EKSA_VSPHERE_PASSWORD` environment variable to the new password and get the base64 encoded string of the password using `echo -n "<YOUR_PASSWORD>" | base64`
- Update secret `vsphere-credentials` under `eksa-system` namespace - Update `password`, `passwordCP`, `passwordCSI` field under data and in `kubectl.kubernetes.io/last-applied-configuration` if annotation exists.
- Modify any field in the cluster config file and then run `eksctl anywhere upgrade cluster -f <cluster-config-file>` to trigger a full cluster upgrade. This will automatically apply the new credentials to all related secrets.

3. Update all vSphere credentials in related Secret objects using [vSphere credential update script](https://github.com/aws/eks-anywhere/blob/main/scripts/update_vsphere_credential.sh) in EKS Anywhere github repository, follow the steps below:
- Set `KUBECONFIG` environment variable to the kubeconfig file generated by EKS Anywhere.

> - Set `EKSA_VSPHERE_USERNAME` and `EKSA_VSPHERE_PASSWORD` with new credentials as environment variables.

Reviewer: ```suggestion
Update `EKSA_VSPHERE_USERNAME` and `EKSA_VSPHERE_PASSWORD` as environment variables with the new credentials.
```

- Save the script and run it by passing EKS Anywhere cluster name and the vSphere server name, `./update_vsphere_credential.sh CLUSTER_NAME VSPHERE_SERVER_NAME`.

> - Note that the script does not update `{CLUSTER_NAME}-csi-vsphere-config`, if vSphere CSI is configured for your cluster, you must manually update password in `{CLUSTER_NAME}-csi-vsphere-config` under `eksa-system` namespace.

Reviewer: ```suggestion
>**_NOTE:_** The script does not update `{CLUSTER_NAME}-csi-vsphere-config`, if vSphere CSI is configured for your cluster, you must manually update password in `{CLUSTER_NAME}-csi-vsphere-config` under `eksa-system` namespace.
```
